jenkins/release-manifest-commit-lock/release-manifest-commit-lock.jenkinsfile

```diff
@@ -0,0 +1,168 @@
+                    if (MANIFEST_LOCK_ACTION.isEmpty() || RELEASE_VERSION.isEmpty() || OPENSEARCH_RELEASE_CANDIDATE.isEmpty() || OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE.isEmpty()) {
+                                currentBuild.result = 'ABORTED'
+                                error('Make sure all the parameters are passed in.')
+                    }
+                }
+            }
+        }
+        stage('Match Build Manifest') {
+            when {
+                expression { params.MANIFEST_LOCK_ACTION == 'Match Build Manifest' }
+            }
+            steps {
+                script {
+                    def updateManifest = { String productName, String releaseCandidate ->
+                        def buildManifestUrl = "https://ci.opensearch.org/ci/dbc/distribution-build-${productName}/${params.RELEASE_VERSION}/${releaseCandidate}/linux/x64/tar/dist/${productName}/manifest.yml"
+                        def buildManifestContent = sh(script: "curl -sSL ${buildManifestUrl}", returnStdout: true).trim()
+                        def buildManifest = readYaml text: buildManifestContent
+                        def existingManifest = readYaml file: "manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml"
+                        def selectedComponents = params.COMPONENTS.isEmpty() ? existingManifest.components : existingManifest.components.findAll { component ->
+                            params.COMPONENTS.split(',').any { it.trim() == component.name }
+                        }
+                        selectedComponents.each { componentName ->
+                            def existingComponent = existingManifest.components.find { it.name == componentName.name }
+                            if (existingComponent) {
+                                def newComponentRef = buildManifest.components.find { it.name == componentName.name }?.ref
+                                if (newComponentRef) {
+                                    existingComponent.ref = newComponentRef
+                                }
+                            }
+                        }
+                        writeYaml file:  "manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml", data: existingManifest, overwrite: true
+                        sh """
+                                    yq eval -i '.' manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml
+                                    sed -i '1s/^/---\\n/' manifests/${params.RELEASE_VERSION}/${productName}-${params.RELEASE_VERSION}.yml
+                        """
+                    }
+                    updateManifest("opensearch", params.OPENSEARCH_RELEASE_CANDIDATE)
+                    updateManifest("opensearch-dashboards", params.OPENSEARCH_DASHBOARDS_RELEASE_CANDIDATE)
```

Comment:
The product can be either or BOTH. Maybe add null check and call based on the provided string value.

Reply:
Its always both Sayali, because the `COMPONENTS` input can be a component of OpenSearch or OpenSearch Dashboards. The logic will infer in which manifest that component exists and update it, hence its always both that will get executed.

Reply:
We can have a 2 inputs componet_os and componet_osd, its not good this way, a user has to know which component in which manifest and also input in right field ( componet_os or componet_osd) and we have again add guardrails if the os component in componet_osd exit and vice versa.

Reply:
So basically we give both RC number and specific component name and it will parse both manifests to find the right fit?

Reply:
Thats true Sayali.

Reply:
you can even have mix of them as well, example `OpenSearch, searchRelevanceDashboards`.

Reply:
Sounds good. Thanks!